### PR TITLE
Updating hello-android to use Kotlin

### PR DESCRIPTION
**Pull request by tanderson-ld**
This PR is to rework the project from a Java Android project to a Kotlin Android project.  Also improvements to the usage of the SDK from a threading standpoint by moving initialization to application startup instead of the UI layer activity.

### CODE REVIEW

**Review by louis-launchdarkly — COMMENTED** [review]
The change itself looks okay, and I double checked it works correctly on Android Studio - and we may need to do something more for this repo if you plan to change this one to hello-android-kotlin:

Apparently, there is a 4.0 branch for the latest example (which should have been rename to main, and the current main rename to 3.x). So I think we should merge this change to the 4.0 branch (as 3.x is still not fully EOL yet).

**Comment by tanderson-ld** [comment]
Let's make those in the hello-android-java repo.

File: app/src/main/java/com/launchdarkly/hello_android/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+        if (BOOLEAN_FLAG_KEY == "my-boolean-flag") {
+            val builder = AlertDialog.Builder(this)
+            builder.setMessage("BOOLEAN_FLAG_KEY was not customized for your account.")
+            builder.create().show()
+        }
+
+        if (LAUNCHDARKLY_MOBILE_KEY == "mobile-key-from-launch-darkly-website") {
+            val builder = AlertDialog.Builder(this)
+            builder.setMessage("LAUNCHDARKLY_MOBILE_KEY was not customized for your account.")
```
Comment (louis-launchdarkly) [inline app/src/main/java/com/launchdarkly/hello_android/MainActivity.kt]: Shall we use `for this application` instead? So the customer is not confused about their LaunchDarkly account?

**Review by louis-launchdarkly — CHANGES_REQUESTED** [review]
Similar comment as the `hello-android-java` one, and the other change looks good.

**Review by louis-launchdarkly — APPROVED** [review]

File: app/src/main/java/com/launchdarkly/hello_android/MainActivity.kt (hunk 1 above)
Comment (tanderson-ld) [reply app/src/main/java/com/launchdarkly/hello_android/MainActivity.kt]: Updated to use your proposed string.  Done

**Pull request merged**